configure: build triple no longer propagates to host and target triples

The makefiles expect that the build triple will always be included in the host and target triples. configure used to make sure this happened but no longer seems to.

If you configure like `../configure --build=x86_64-unknown-linux-gnu --host=i686-unknown-linux-gnu` configure outputs

```
configure: CFG_HOST             := i686-unknown-linux-gnu 
configure: CFG_TARGET           := i686-unknown-linux-gnu 
```

And make results in some weirdness:

```
cfg: build triple x86_64-unknown-linux-gnu
cfg: host triples i686-unknown-linux-gnu
cfg: target triples i686-unknown-linux-gnu
cfg: non-build host triples i686-unknown-linux-gnu
cfg: non-build target triples i686-unknown-linux-gnu
cfg: enabling more debugging (CFG_ENABLE_DEBUG)
cfg: host for i686-unknown-linux-gnu is i386
cfg: os for i686-unknown-linux-gnu is unknown-linux-gnu
cfg: good valgrind for i686-unknown-linux-gnu is 1
cfg: using CC=gcc (CFG_CC)
cfg: enabling valgrind run-pass tests (CFG_ENABLE_VALGRIND_RPASS)
cfg: no lualatex found, deferring to xelatex
cfg: no xelatex found, deferring to pdflatex
cfg: no pdflatex found, disabling LaTeX docs
cfg: no pandoc found, omitting PDF and EPUB docs
cfg: no llnextgen found, omitting grammar-verification
fetch: /rustc
determined most recent snapshot: rust-stage0-2014-10-22-d44ea72-linux-x86_64-3e04d8197a96b0c858e4e2763b3893df35ae2fb3.tar.bz2
got download with ok hash
opening snapshot dl/rust-stage0-2014-10-22-d44ea72-linux-x86_64-3e04d8197a96b0c858e4e2763b3893df35ae2fb3.tar.bz2
removing x86_64-unknown-linux-gnu/stage0/bin/rustc
extracting rust-stage0/bin/rustc
touch: cannot touch ‘/rustc’: Permission denied
```

or maybe (if you already have the snap)

```
$ make -j4 rustc-stage2
cfg: build triple x86_64-apple-darwin
cfg: host triples i686-apple-darwin
cfg: target triples i686-apple-darwin arm-linux-androideabi
cfg: non-build host triples i686-apple-darwin
cfg: non-build target triples i686-apple-darwin arm-linux-androideabi
cfg: enabling more debugging (CFG_ENABLE_DEBUG)
cfg: host for i686-apple-darwin is i386
cfg: host for arm-linux-androideabi is arm
cfg: os for i686-apple-darwin is apple-darwin
cfg: os for arm-linux-androideabi is linux-androideabi
cfg: good valgrind for i686-apple-darwin is
cfg: good valgrind for arm-linux-androideabi is
cfg: using CC=clang (CFG_CC)
cfg: using CXX=clang++ (CFG_CXX)
cfg: enabling valgrind run-pass tests (CFG_ENABLE_VALGRIND_RPASS)
cfg: no pandoc found, omitting PDF and EPUB docs
cfg: no llnextgen found, omitting grammar-verification
make: *** No rule to make target `rustc-stage2-H-x86_64-apple-darwin', needed by `rustc-stage2'. Stop.
```

--- a/configure
+++ b/configure
@@ -1037,6 +1037,14 @@ for target_file in ${CFG_SRC_DIR}mk/cfg/*.mk; do
   CFG_SUPPORTED_TARGET="${CFG_SUPPORTED_TARGET} $(basename "$target_file" .mk)"
 done
 
+# copy build-triples to host-triples so that builds are a subset of hosts
+V_TEMP=""
+for i in $CFG_BUILD $CFG_HOST;
+do
+   echo "$V_TEMP" | grep -qF $i || V_TEMP="$V_TEMP${V_TEMP:+ }$i"
+done
+CFG_HOST=$V_TEMP
+
 # copy host-triples to target-triples so that hosts are a subset of targets
 V_TEMP=""
 for i in $CFG_HOST $CFG_TARGET;

